src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.impl.BulldozerProducerImpl;
+import core.mate.academy.service.impl.ExcavatorProducerImpl;
+import core.mate.academy.service.impl.TruckProducerImpl;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            MachineProducer<Bulldozer> producer = new BulldozerProducerImpl();
+            return (List<T>) producer.get();
+        } else if (type == Excavator.class) {
+            MachineProducer<Excavator> producer = new ExcavatorProducerImpl();
+            return (List<T>) producer.get();
+        } else if (type == Truck.class) {
+            MachineProducer<Truck> producer = new TruckProducerImpl();
+            return (List<T>) producer.get();
+        } else {
+            return new ArrayList<>();
```

Comment:
Instead of returning a new instance of ArrayList directly, it's better to return Collections.emptyList() to avoid unnecessary object creation when no producer has been found.